### PR TITLE
Trim field input when configuring pipeline server

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
JIRA: https://issues.redhat.com/browse/RHOAIENG-2317

## Description

Trim all the fields of ObjectStorage when configuring a pipeline server to make sure the DSPA and secret creation is correct.

## How Has This Been Tested?

1. Configure a pipeline server, input some spaces before and after all the field inputs
2. Configure the server
3. View the server configuration, make sure all the fields are correct and without the spaces at the beginning and end.

## Test Impact

N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by manaswinidas — APPROVED** [review]
/lgtm

Works fine.

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/ConfigurePipelinesServerModal.tsx [hunk 1]
```diff
@@ -64,7 +64,10 @@ export const ConfigurePipelinesServerModal: React.FC<ConfigurePipelinesServerMod
 
   const submit = () => {
     const objectStorage: PipelineServerConfigType['objectStorage'] = {
-      newValue: config.objectStorage.newValue,
+      newValue: config.objectStorage.newValue.map((entry) => ({
+        key: entry.key,
```
Comment (gitdallas) [inline frontend/src/concepts/pipelines/content/configurePipelinesServer/ConfigurePipelinesServerModal.tsx:68]: would `{ ...entry, value: entry.value.trim() }` be good just to show nothing is being done except trimming the value?

**Review by gitdallas — APPROVED** [review]
lgtm, one question

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/ConfigurePipelinesServerModal.tsx (hunk 1 above)
Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/ConfigurePipelinesServerModal.tsx:68]: I like it, please make this change @DaoDaoNoCode

**Review by manaswinidas — COMMENTED** [review]
/lgtm

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm
/approve

tested and trims. and does not allow to submit when only spaces are used

**Pull request merged**